AstSemantics.md

```diff
@@ -111,6 +112,9 @@ Integer loads can specify a *storage size* which is smaller than the result type
 well as a signedness which determines whether the bytes are sign- or zero-
 extended into the result type.
 
+Each `load` and `store` specifies the [linear memory index](Modules.md#index-spaces)
+to operate on with an unsigned integer immediate.
+
```

Comment:
@lukewagner I don't think segmented memory is 'something we definitely want in the long-term', quite the opposite I think it is a place wasm should never go. I'd like the wasm linear memory to be very fast to access, and I see segments compromising this goal. If you have a solution to the memory access performance challenge then that might change my mind?

The use case of accessing JS objects from wasm can use the GC object support, which can use the object pointer as the base and do inline bounds checking, and it's not clear that this would be any slower than accessing segments. A JS canvas etc is not going to be allocated in a large chunk of VM just to avoid bounds checks, so segments pointing to its raw data are going to be using a base and explicit bounds checks anyway?

What optimizations do imported segments support beyond what would be possible with GC object support anyway?

The only optimization opportunity that comes to mind is baking in an offset on the x86, but the base would probably be in a register on other architectures anyway.

I don't see any need to reserve a 0/1 field in operators now either. The common case will be using a single segment so an opcode dedicated to this is useful anyway now, and new opcodes with a segment field could be introduced in future to support segments.